pkg/controller/workload/workload_cache.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package workload
+
+import (
+	"sync"
+
+	workloadapi "kmesh.net/kmesh/api/v2/workloadapi"
+	nets "kmesh.net/kmesh/pkg/nets"
+)
+
+var (
+	WorkloadCache = newWorkloadStore()
+)
+
+type NetworkAddress struct {
+	Network string
+	Address uint32
+}
+
+type workloadStore struct {
+	byUid  map[string]*workloadapi.Workload
+	byAddr map[NetworkAddress]*workloadapi.Workload
+	mutex  sync.RWMutex
+}
+
+func newWorkloadStore() *workloadStore {
+	return &workloadStore{
+		byUid:  make(map[string]*workloadapi.Workload),
+		byAddr: make(map[NetworkAddress]*workloadapi.Workload),
+	}
+}
+
+func (w *workloadStore) GetWorkloadByUid(uid string) *workloadapi.Workload {
+	w.mutex.RLock()
+	defer w.mutex.RUnlock()
+	return w.byUid[uid]
+}
+
+func (w *workloadStore) GetWorkloadByAddr(networkAddress NetworkAddress) *workloadapi.Workload {
+	w.mutex.RLock()
+	defer w.mutex.RUnlock()
+	return w.byAddr[networkAddress]
+}
+
+func composeNetworkAddress(network string, addr uint32) NetworkAddress {
+	networkAddress := NetworkAddress{
+		Network: network,
+		Address: addr,
+	}
+
+	return networkAddress
+}
+
+func (w *workloadStore) addWorkload(workload *workloadapi.Workload) {
+	uid := workload.Uid
+
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	workloadByUid, exist := w.byUid[uid]
+	if exist {
```

Comment:
we can optimize a little bit more, avoid operating the store when the old workload equals the new one

Reply:
Need to determine if networkAddress are equal? This way, it seems more cumbersome and requires a double-layer loop to compare networkAddress one by one.

Reply:
I think we can use proto.Equal

Reply:
Workloads stored by networkAddress need to be cross compared to determine whether they are equal or residual, and IP traversal is required to store new workloads by networkAddress. Therefore, even if proto.Equal is used, there will inevitably be some complex or repetitive actions.

Reply:
Not sure i get your point. My suggestion is to skip all below processing when workload equals

Reply:
Sorry, I have made this problem more complicated. Fixed it.